xscen/finalize.py

```diff
@@ -108,25 +116,56 @@ def clean_up(
         eg. {'global': {'title': 'amazing new dataset'}, 'tasmax': {'note': 'important info about tasmax'}}
     change_attr_prefix: str
         Replace "cat/" in the catalogue global attrs by this new string
+    to_level: str
+        The processing level to assign to the output.
 
     Returns
     -------
     ds: xr.Dataset
         Cleaned up dataset
     """
-    if var_and_convert_units:
-        logger.info(f"Converting units: {var_and_convert_units}")
-        ds = change_units(ds=ds, variables_and_units=var_and_convert_units)
+
+    if variables_and_units:
+        logger.info(f"Converting units: {variables_and_units}")
+        ds = change_units(ds=ds, variables_and_units=variables_and_units)
 
     # unstack nans
     if maybe_unstack_dict:
         ds = maybe_unstack(ds, **maybe_unstack_dict)
 
-    # put back feb 29th
-    if add_feb_29:
-        logging.info("Adding back february 29th by linear interpolation")
-        with_missing = convert_calendar(ds, "standard", missing=np.NaN)
-        ds = with_missing.interpolate_na("time", method="linear")
+    # convert calendar
+    if convert_calendar_kwargs:
+        ds_copy = ds.copy()
+
+        # if missing_by_var exist make sure missing data are added to time axis
+        if missing_by_var:
+            convert_calendar_kwargs["missing"] = np.nan
```

Comment:
```suggestion
            convert_calendar_kwargs.setdefault("missing", np.nan)
```
Comme ça on peut passer un `missing` autre que np.nan pour tout sauf les entrées de  `missing_by_var`.